Octokit.Tests/Clients/UsersClientTests.cs

```diff
@@ -92,7 +92,7 @@ public void SendsUpdateToCorrectUrl()
 
         public class SerializationTests
         {
-            [Fact]
+            [Fact(Skip = "Change to serializer means null is now returned")]
```

Comment:
@shiftkey can you shed any light on the behaviour this test is asserting?  It seems that when you added support for Organizatino/User "type" on Account, you made it nullable and then have this test here to prove that it's null when deserializing a json fragment that is not a user/org/account at all.

Meanwhile in this PR I found that the deserializer instantiating the requested class even when the json had no matching fields seemed to go against what other parts of the code were expecting (linked in PR body) so i changed it to only instantiate class when it finds at least 1 expected field in the json, otherwise it returns a null object.  This test here is the only place (we have test coverage for at least) that seems to be impacted by this.